docs/development/module/off-chain-workers.md

```diff
@@ -0,0 +1,402 @@
+$ curl -X POST -vk 'http://localhost:9933' -H "Content-Type:application/json;charset=utf-8" \
+  -d '{
+    "jsonrpc":2.0,
+    "id":1,
+    "method":"author_insertKey",
+    "params": [
+      "<YourKeyTypeId>",
+      "<YourSeedPhrase>",
+      "<YourPublicKey>"
+    ]
+  }'
+```
+
+A new key is now added in the local keystore.
+
+## Signed Transactions
+
+Now you are ready to to make a signed transaction from the off-chain worker. Go back to your pallet
+in `my_offchain_worker.rs`.
+
+```rust
+decl_module! {
+  pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+    // --snip--
+
+    // This is your regular on-chain extrinsics
+    pub fn onchain_callback(origin, _block: T::BlockNumber, input: Vec<u8>) -> dispatch::Result {
+      let who = ensure_signed(origin)?;
+      debug::info!("{:?}", core::str::from_utf8(&input).unwrap());
+      Ok(())
+    }
+
+    fn offchain_worker(block: T::BlockNumber) {
+      // Here we specify the function to be called back on-chain in next block import.
+      let call = Call::onchain_callback(block, b"hello world!".to_vec());
+      T::SubmitSignedTransaction::submit_signed(call);
+    }
+  }
+}
+```
+
+After having defined the on-chain callback function, in the off-chain worker you can specify
+that function to be called back in the next block import phase. You then submit a signed transaction
+to the node.
+
+If you look at the implementation of `fn system::offchain::submit_signed` in the
+[Substrate codebase](https://github.com/paritytech/substrate/blob/a98625501be68cc3084e666497c16b111741dded/frame/system/src/offchain.rs#L106-L115),
+you will see it is calling the on-chain callback for each key in the local keystore. But since
+you only have one key in the local keystore now, you are calling the function only once.
+
+## Unsigned Transactions
+
+With the following code, you are able to send an unsigned transaction back to the chain.
+
+```rust
+decl_module! {
+  pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+    // --snip--
+
+    // This is your regular on-chain extrinsics
+    pub fn onchain_callback(_origin, _block: T::BlockNumber, input: Vec<u8>) -> dispatch::Result {
+      debug::info!("{:?}", core::str::from_utf8(&input).unwrap());
+      Ok(())
+    }
+
+    fn offchain_worker(block: T::BlockNumber) {
+      // Here we specify the function to be called back on-chain in next block import.
+      let call = Call::onchain_callback(block, b"hello world!".to_vec());
+      T::SubmitUnsignedTransaction::submit_unsigned(call);
+    }
+  }
+}
+```
+
+By default, all unsigned transactions are treated as invalid transactions. You need to add the
+following code piece in `my_offchain_worker.rs` to explicitly allow some on-chain functions to call
+unsigned transactions.
+
+```rust
+decl_module! {
+  // --snip--
+}
+
+impl<T: Trait> Module<T> {
+  // --snip--
+}
+
+#[allow(deprecated)]
+impl<T: Trait> support::unsigned::ValidateUnsigned for Module<T> {
```

Comment:
This should be removed ASAP, including the paragraph that follows this code snippet. Unfortunately, it seems we don't actually have a code example of using `SignedExtension` to do this. At a minimum, we should like people to the conceptual-extrinsics doc where we talk about validate-unsigned.

I think we can probably merge now with the example so that we don't leave people hanging, but should really caution them with "don't make this a habit."

Reply:
Updated, and added links at the bottom of each section, and the last `Next Steps` section.